### PR TITLE
Call applyBlockDefaults from addBlock/insertBlock, add initialValue() configuration option for blocks

### PR DESCRIPTION
**Pull request by tiberiuichim**

### CODE REVIEW

**Comment by tiberiuichim** [comment]
@sneridagh @davisagli 

Here's the proposal: we add a new `.initialValue` function to the block configuration, which can be called like this:

```
    value = blocksConfig[type].initialValue({
      id,
      value,
      formData: newFormData,
    });
```

Thoughs on naming and function signature?

File: docs/source/blocks/settings.md [hunk 1]
```diff
@@ -137,6 +137,26 @@ defineMessages({
 
 Our new block should be ready to use in the editor.
 
+## Common block options
+
+It is a common pattern to use the block configuration to allow customization of a block's behavior or to provide block-specific implementation of various Volto mechanisms. Some of these "common" options are:
```
Comment (stevepiercy) [inline docs/source/blocks/settings.md:142]:
```suggestion
It is a common pattern to use the block configuration to allow customization of a block's behavior or to provide block-specific implementation of various Volto mechanisms.
Some of these common options are described in the following sections.
```

File: docs/source/blocks/settings.md [hunk 2]
```diff
@@ -137,6 +137,26 @@ defineMessages({
...
+
+### `blockHasValue`
+
+Returns true if the provided block data represents a value for the current block. Required for alternate default block types implementations. See also [Settings reference](/configuration/settings-reference)
```
Comment (stevepiercy) [inline docs/source/blocks/settings.md:146]:
````suggestion
`blockHasValue` returns `true` if the provided block data represents a value for the current block.
Required for alternate default block types implementations.

```{seealso}
See also [Settings reference](/configuration/settings-reference).
```
````

File: docs/source/blocks/settings.md [hunk 3]
```diff
@@ -137,6 +137,26 @@ defineMessages({
...
+
+### `initialValue`
+
+A function that can be used to get the initial value for a block. It has the
```
Comment (stevepiercy) [inline docs/source/blocks/settings.md:150]:
```suggestion
`initialValue` is a function that can be used to get the initial value for a block.
```

File: docs/source/blocks/settings.md [hunk 4]
```diff
@@ -137,6 +137,26 @@ defineMessages({
...
+following signature:
```
Comment (stevepiercy) [inline docs/source/blocks/settings.md:151]:
```suggestion
It has the following signature.
```

File: docs/source/blocks/settings.md [hunk 5]
```diff
@@ -137,6 +137,26 @@ defineMessages({
...
+
+`initialValue({id, value, formData, intl}) => newFormData`
```
Comment (stevepiercy) [inline docs/source/blocks/settings.md:153]:
````suggestion
```jsx
initialValue({id, value, formData, intl}) => newFormData
```
````

File: docs/source/blocks/settings.md [hunk 6]
```diff
@@ -137,6 +137,26 @@ defineMessages({
...
+
+### `blockSchema`
+
+A must-have for modern Volto blocks, this is a function (or directly the schema object) that returns the schema for the block data. While it's not required, having the schema defined enables the block to have its initial value based on the default values declared in the schema.
```
Comment (stevepiercy) [inline docs/source/blocks/settings.md:157]:
```suggestion
A must-have for modern Volto blocks, `blockSchema` is a function, or directly the schema object, that returns the schema for the block data.
Although it's not required, defining the schema enables the block to have its initial value based on the default values declared in the schema.
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Some MyST enhancements, formatting, and grammar fixes. Thanks for the addition!

**Comment by davisagli** [comment]
@tiberiuichim btw, each time you press the "commit suggestion" button, it creates a new commit that triggers all the CI workflows. You can avoid this by going to the "Files changed" tab, where you have the option to "Add suggestion to batch" and then make a single commit with multiple suggestions included.

**Review by stevepiercy — APPROVED** [review]
Looks pretty! https://deploy-preview-5320--volto.netlify.app/blocks/settings.html#common-block-options

**Review by sneridagh — APPROVED** [review]

**Comment by sneridagh** [comment]
@tiberiuichim could you please add the changelog if it's ready? I would merge it right away!

**Comment by sneridagh** [comment]
@tiberiuichim ping!

**Pull request merged**